martin-mbtiles/src/mbtiles.rs

```diff
@@ -280,13 +281,64 @@ impl Mbtiles {
     where
         for<'e> &'e mut T: SqliteExecutor<'e>,
     {
-        if is_deduplicated_type(&mut *conn).await? {
-            Ok(MbtType::DeDuplicated)
+        let mbt_type = if is_deduplicated_type(&mut *conn).await? {
+            MbtType::DeDuplicated
         } else if is_tile_tables_type(&mut *conn).await? {
-            Ok(MbtType::TileTables)
+            MbtType::TileTables
         } else {
-            Err(MbtError::InvalidDataFormat(self.filepath.clone()))
+            return Err(MbtError::InvalidDataFormat(self.filepath.clone()));
+        };
+
+        self.check_for_uniqueness_constraint(&mut *conn, &mbt_type)
+            .await?;
+
+        Ok(mbt_type)
+    }
+
+    async fn check_for_uniqueness_constraint<T>(
+        &self,
+        conn: &mut T,
+        mbt_type: &MbtType,
+    ) -> MbtResult<()>
+    where
+        for<'e> &'e mut T: SqliteExecutor<'e>,
+    {
+        let table_name = match mbt_type {
+            MbtType::TileTables => "tiles",
+            MbtType::DeDuplicated => "map",
+        };
+
+        let index_query =
+            format!("SELECT name FROM pragma_index_list('{table_name}') WHERE [unique] = 1;");
+        let indexes = query(&index_query).fetch_all(&mut *conn).await?;
+
+        // Ensure there is some index on tiles that has a unique constraint on (zoom_level, tile_row, tile_column)
+        for index in indexes {
+            let mut unique_idx_cols = HashSet::new();
+            let key_column_query = format!(
+                "SELECT DISTINCT name FROM pragma_index_info('{}')",
+                index.get::<String, _>("name")
```

Comment:
this is a minor SQL injection attack - using user-provided data (column name in many databases can be arbitrary, possibly in sqlite too, or could be in the future).  Please use `query().bind()` or some other similar approach to pass in the query argument.